code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -2262,10 +2262,39 @@ All effects don't start immediately, but rather get worse over time; the rate is
 
 	..()
 
+
+
 /datum/reagent/consumable/ethanol/plasmaflood/on_mob_end_metabolize(mob/living/L)
 	to_chat(L, "<span class='warning'>You no longer feel immune to burning!</span>")
 	. = ..()
 
+/datum/reagent/consumable/ethanol/beeburst
+	name = "Bee Burst"
+	glass_name = "Bee Burst"
+	description = "Its un-bee-lievable how much honey is in here! (A small warning on the bottom warns of bees appearing)"
+	glass_desc = "Its un-bee-lievable how much honey is in here! (A small warning on the bottom warns of bees appearing)"
+	color = "#f0eb89"  // rgb: 240, 235, 137
+	boozepwr = 70
+	quality = DRINK_NICE
+	taste_description = "a bee hive"
+	glass_icon_state = "beeburst"
+	//glass_icon_state = "plasmaflood"
+
+/datum/reagent/consumable/ethanol/beeburst/on_mob_metabolize(mob/living/L)
+	to_chat(L, "<span class='notice'>You feel fluttering in your stomach..</span>")
+	. = ..()
+
+/datum/reagent/consumable/ethanol/beeburst/on_mob_life(mob/living/carbon/M)
+	if(prob(40))  // add delay between spawns? 
```

Comment:
this will make... a lot of bees

Reply:
not sure if cooldown macros exist (ctrl + f "COOLDOWN_DECLARE" to see if it shows anything but otherwise do this

```c
/obj
    var/cooldown

/obj/proc/activate_cooldown()
    if(!cooldown || (cooldown + 10 SECONDS < world.time))
        cooldown = world.time
```
which will create a ten second cooldown

Reply:
so how would i check the cooldown during the execution and make it not run